lib/membrane_mp4/demuxer/isom/samples_info.ex

```diff
@@ -63,10 +66,13 @@ defmodule Membrane.MP4.Demuxer.ISOM.SamplesInfo do
   end
 
   defp do_get_samples(samples_info, data, buffers) do
-    [%{size: size, track_id: track_id} = sample | samples] = samples_info.samples
+    [%{size: size, track_id: track_id, sample_offset: sample_offset} = sample | samples] =
+      samples_info.samples
 
-    if size <= byte_size(data) do
-      <<payload::binary-size(size), rest::binary>> = data
+    to_skip = sample_offset - samples_info.mdat_iterator
+
+    if to_skip + size <= byte_size(data) do
+      <<_to_skip::binary-size(to_skip), payload::binary-size(size), rest::binary>> = data
```

Comment:
It would look better as a `case`

```suggestion
    case data do
      <<_to_skip::binary-size(to_skip), payload::binary-size(size), rest::binary>> ->
```

Reply:
done